File: app/src/main/java/com/frogobox/evpn/activity/VPNListActivity.java
package com.frogobox.evpn.activity;

import android.content.Intent;

import android.graphics.PorterDuff;
import android.graphics.drawable.Drawable;
import android.os.Bundle;
import androidx.appcompat.app.ActionBar;
import androidx.appcompat.widget.Toolbar;
import android.view.MenuItem;
import android.view.View;
import android.widget.AdapterView;
import android.widget.ImageView;
import android.widget.ListView;
import android.widget.TextView;

import com.frogobox.evpn.R;
import com.frogobox.evpn.adapter.ServerListAdapter;
import com.frogobox.evpn.model.Server;
import com.google.android.gms.ads.AdListener;
import com.google.android.gms.ads.AdRequest;
import com.google.android.gms.ads.AdView;
import com.google.android.gms.ads.InterstitialAd;
import com.google.android.gms.ads.MobileAds;

import java.util.List;

import de.blinkt.openvpn.core.VpnStatus;

public class VPNListActivity extends BaseActivity {
    private ListView listView;
    private ServerListAdapter serverListAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_vpnlist);
        MobileAds.initialize(this, String.valueOf(R.string.admob_publisher_id));

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbarr);
        setSupportActionBar(toolbar);
        ActionBar actionBar = getSupportActionBar();
        actionBar.setDisplayHomeAsUpEnabled(true);
        actionBar.setDisplayShowHomeEnabled(true);
        final Drawable upArrow = getResources().getDrawable(R.drawable.ic_arrow_back_black_24dp);
        upArrow.setColorFilter(getResources().getColor(android.R.color.white), PorterDuff.Mode.SRC_ATOP);
        actionBar.setHomeAsUpIndicator(upArrow);
        AdView mAdMobAdView = (AdView) findViewById(R.id.admob_adview);
        AdRequest adRequest = new AdRequest.Builder()
                .build();
        mAdMobAdView.loadAd(adRequest);

        mAdMobAdView.setAdListener(new AdListener() {
            @Override
            public void onAdLoaded() {
            }

            @Override
            public void onAdFailedToLoad(int errorCode) {
                AdRequest adRequest = new AdRequest.Builder().build();
                mAdMobAdView.loadAd(adRequest);
            }

            @Override
            public void onAdOpened() {
            }

            @Override
            public void onAdLeftApplication() {
                AdRequest adRequest = new AdRequest.Builder().build();
                mAdMobAdView.loadAd(adRequest);
            }

            @Override
            public void onAdClosed() {
            }
        });


        final InterstitialAd mInterstitial = new InterstitialAd(this);
        mInterstitial.setAdUnitId(getString(R.string.admob_interstitial));
        mInterstitial.loadAd(new AdRequest.Builder().build());
        mInterstitial.setAdListener(new AdListener() {
            @Override
            public void onAdLoaded() {
                
                super.onAdLoaded();
                if (mInterstitial.isLoaded()) {
                    mInterstitial.show();
                }
            }
        });

        if (!VpnStatus.isVPNActive())
            connectedServer = null;

        listView = (ListView) findViewById(R.id.list);
    }

    @Override
    protected void onResume() {
        super.onResume();

        invalidateOptionsMenu();

        buildList();
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        if (item.getItemId() == android.R.id.home) {
            finish();
            return true;
        }
        return super.onOptionsItemSelected(item);
    }

    @Override
    protected void ipInfoResult() {
        serverListAdapter.notifyDataSetChanged();
    }

    private void buildList() {
        String country = getIntent().getStringExtra(MainActivity.EXTRA_COUNTRY);
        final List<Server> serverList = dbHelper.getServersByCountryCode(country);
        serverListAdapter = new ServerListAdapter(this, serverList);

        TextView countryname = (TextView) findViewById(R.id.elapse);
        countryname.setText(country);

        String code = getIntent().getStringExtra(MainActivity.EXTRA_COUNTRY).toLowerCase();
        if (code.equals("do"))
            code = "dom";

        ((ImageView) findViewById(R.id.imgv))
                .setImageResource(
                        getResources().getIdentifier(code,
                                "drawable",
                                getPackageName()));

        listView.setAdapter(serverListAdapter);
        listView.setOnItemClickListener(new AdapterView.OnItemClickListener() {
            @Override
            public void onItemClick(AdapterView<?> parent, View view, int position, long id) {
                Server server = serverList.get(position);
                BaseActivity.sendTouchButton("detailsServer");
                Intent intent = new Intent(VPNListActivity.this, VPNInfoActivity.class);
                intent.putExtra(Server.class.getCanonicalName(), server);
                VPNListActivity.this.startActivity(intent);
            }
        });

        getIpInfo(serverList);
    }
}

File: app/src/main/java/com/frogobox/vpnhero/view/ui/activity/VPNListActivity.kt
package com.frogobox.vpnhero.view.ui.activity

import android.os.Bundle
import androidx.recyclerview.widget.LinearLayoutManager
import com.frogobox.vpnhero.R
import com.frogobox.vpnhero.base.adapter.BaseViewListener
import com.frogobox.vpnhero.base.ui.BaseActivity
import com.frogobox.vpnhero.helper.Constant.Variable.EXTRA_COUNTRY
import com.frogobox.vpnhero.source.model.Server
import com.frogobox.vpnhero.view.adapter.ServerViewAdapter
import de.blinkt.openvpn.core.VpnStatus
import kotlinx.android.synthetic.main.activity_vpnlist.*

class VPNListActivity : BaseActivity(), BaseViewListener<Server> {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_vpnlist)
        setupAdsMonetize()
        setupComponentView()
    }

    private fun setupAdsMonetize() {
        setupShowAdsBanner(findViewById(R.id.admob_adview))
        setupShowAdsInterstitial()
    }

    private fun setupComponentView() {
        val country = intent.getStringExtra(EXTRA_COUNTRY)
        if (!VpnStatus.isVPNActive()) connectedServer = null
        setupDetailActivity("")
        setupRecyclerView(country)
    }

    private fun setupRecyclerView(country: String) {
        val serverList = dbHelper.getServersByCountryCode(country)
        val serverViewAdapter = ServerViewAdapter()
        serverViewAdapter.setupRequirement(this, this, serverList, R.layout.view_item_server)
        recycler_view.layoutManager = LinearLayoutManager(this, LinearLayoutManager.VERTICAL, false)
        recycler_view.adapter = serverViewAdapter
        getIpInfoFromServerList(serverList)
    }

    override fun onItemClicked(data: Server) {
        baseStartActivity<VPNInfoActivity, Server>(Server::class.java.canonicalName!!, data)
    }

    override fun onItemLongClicked(data: Server) {}

}